controller/iteration_blackbox_test.go

```diff
@@ -763,3 +765,259 @@ func assertChildIterationLinking(t *testing.T, target *app.Iteration) {
+		wis, err := rest.db.WorkItems().LoadByIteration(svc.Context, childIteration.ID)
+		require.Nil(t, err)
+		assert.Empty(t, wis)
+
+		// parent should get more 3 WI
+		parentIteration := fxt.IterationByName("parent")
+		wis, err = rest.db.WorkItems().LoadByIteration(svc.Context, parentIteration.ID)
+		require.Nil(t, err)
+		// first iteration already have 3 & 3 more from child iteration
+		assert.Len(t, wis, 3+3)
+
+		// verify that root iteration still does not have any WI
+		rootIteration := fxt.IterationByName("root")
+		wis, err = rest.db.WorkItems().LoadByIteration(svc.Context, rootIteration.ID)
+		require.Nil(t, err)
+		assert.Empty(t, wis)
+	})
+
+	// Following test creates the structure shown in diagram
+	// root Iteration
+	// |___________Iteration 1 (5 WI)
+	// |                |___________Iteration 2 (5 WI)
+	// |                                |___________Iteration 3 (5 WI)
+	// |___________Iteration 4 (2 WI)
+	//                     |___________Iteration 5 (3 WI)
+	// then deletes iteration1 & iteration5 to verify the effect
+	rest.T().Run("success - verify that workitems are updated correctly", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Iterations(6,
+				func(fxt *tf.TestFixture, idx int) error {
+					i := fxt.Iterations[idx]
+					switch idx {
+					case 1:
+						i.MakeChildOf(*fxt.Iterations[0])
+					case 2:
+						i.MakeChildOf(*fxt.Iterations[1])
+					case 3:
+						i.MakeChildOf(*fxt.Iterations[2])
+					case 4:
+						i.MakeChildOf(*fxt.Iterations[0])
+					case 5:
+						i.MakeChildOf(*fxt.Iterations[4])
+					}
+					return nil
+				}),
+			tf.WorkItems(20, func(fxt *tf.TestFixture, idx int) error {
+				wi := fxt.WorkItems[idx]
+				switch idx {
+				case 0, 1, 2, 3, 4:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+				case 5, 6, 7, 8, 9:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[2].ID.String()
+				case 10, 11, 12, 13, 14:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[3].ID.String()
+				case 15, 16:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[4].ID.String()
+				case 17, 18, 19:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[5].ID.String()
+				}
+				return nil
+			}))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		iterationToDelete := fxt.Iterations[1]
+		test.DeleteIterationNoContent(t, svc.Context, svc, ctrl, iterationToDelete.ID)
+		wis, err := rest.db.WorkItems().LoadByIteration(svc.Context, iterationToDelete.ID)
+		require.Nil(t, err)
+		assert.Empty(t, wis)
+
+		// Verify that 15 WIs are moved to Root iteration
+		wis, err = rest.db.WorkItems().LoadByIteration(svc.Context, fxt.Iterations[0].ID)
+		require.Nil(t, err)
+		assert.Len(t, wis, 15)
+
+		iterationToDelete = fxt.Iterations[5]
+		test.DeleteIterationNoContent(t, svc.Context, svc, ctrl, iterationToDelete.ID)
+		wis, err = rest.db.WorkItems().LoadByIteration(svc.Context, iterationToDelete.ID)
```

Comment:
IMHO this (`LoadByIteration`) must return a not found error instead of an empty slice when the `iterationToDelete` no longer exists.